meeting_scheduler/src/rest/resources.py

```diff
@@ -116,10 +127,15 @@ def delete(self, meeting_id: int):
 
 class TimeslotApi(Resource):
     timeslot_schema = TimeslotSchema()
+    request_schema = RequestSchema()
     timeslot_db_service = CRUDService(Timeslot, db)
 
     def get(self, timeslot_id: int = None):
         if timeslot_id is None:
+            if request.args:
+                req = self.request_schema.get_request(request.args)
+                if req:
+                    return get_free_timeslots(req), 200
```

Comment:
too many ifs to my taste ( and repeatable pattern in few endpoints). 
can we make it generic, e.g. by subclassing Resource and inheriting from it?